pkg/zcapld/create.go

```diff
@@ -44,6 +44,18 @@ type CapabilityOptions struct {
 	Challenge        string
 	Domain           string
 	CapabilityChain  []interface{}
+	Caveats          []Caveat
+}
+
+// Caveat types.
+const (
+	CaveatTypeExpiry = "expiry"
+)
+
+// Caveat is a restriction on how capability may be used.
+type Caveat struct {
```

Comment:
I think caveats should be a generic mechanism - there will be several types of caveats, not just expiry.

Can you please leave a TODO to generalize this mechanism later?

Reply:
ok, I will. But it seems like we just need to extend a structure.
e.g
```
type Caveat struct {
	Type     string `json:"type"`
	Duration uint64 `json:"duration"`
        URI uint64 `json:"uri"`
}
```

Reply:
@llorllale Done!

Reply:
> ok, I will. But it seems like we just need to extend a structure.

Which is why I think a better way would be to base this on maps. The only common property will be "type". (and "id" if we were to model this strictly in JSONLD terms, but I have other thoughts about that...)